components/tidb_query/src/rpn_expr/impl_json.rs

```diff
@@ -17,6 +17,67 @@ fn json_type(arg: &Option<Json>) -> Result<Option<Bytes>> {
         .map(|json_arg| Bytes::from(json_arg.json_type())))
 }
 
+#[rpn_fn(raw_varg, min_args = 2, extra_validator = json_modify_validator)]
+#[inline]
+fn json_set(args: &[ScalarValueRef]) -> Result<Option<Json>> {
+    json_modify(args, ModifyType::Set)
+}
+
+#[rpn_fn(raw_varg, min_args = 2, extra_validator = json_modify_validator)]
+#[inline]
+fn json_insert(args: &[ScalarValueRef]) -> Result<Option<Json>> {
+    json_modify(args, ModifyType::Insert)
+}
+
+#[rpn_fn(raw_varg, min_args = 2, extra_validator = json_modify_validator)]
+#[inline]
+fn json_replace(args: &[ScalarValueRef]) -> Result<Option<Json>> {
+    json_modify(args, ModifyType::Replace)
+}
+
+#[inline]
+fn json_modify(args: &[ScalarValueRef], mt: ModifyType) -> Result<Option<Json>> {
+    assert!(args.len() >= 2);
+    // base Json argument
+    let base: &Option<Json> = args[0].as_ref();
+    let mut base = base.as_ref().map_or(Json::None, |json| json.to_owned());
+
+    let buf_size = args.len() / 2;
+
+    let mut path_expr_list = Vec::with_capacity(buf_size);
+    let mut values = Vec::with_capacity(buf_size);
+
+    for chunk in args[1..].chunks(2) {
```

Comment:
If you prefer a more functional style, here we apply a `fold` combinator.

Reply:
> If you prefer a more functional style, here we apply a `fold` combinator.

Well, I think using for loop is enough :( 
But thanks a lot~